src/libraries/Microsoft.Extensions.Configuration/src/ConfigurationProvider.cs

```diff
@@ -95,7 +95,7 @@ public virtual void Load()
 
         private static string Segment(string key, int prefixLength)
         {
-            int indexOf = key.IndexOf(ConfigurationPath.KeyDelimiter, prefixLength, StringComparison.OrdinalIgnoreCase);
+            int indexOf = key.IndexOf(ConfigurationPath.KeyDelimiter, prefixLength, StringComparison.Ordinal);
```

Comment:
Changing to Ordinal makes additional assumptions about what KeyDelimiter is.  As long as we're making such an assumption, we could also assume (and assert) that it's a single character, and just IndexOf for that, e.g.
```C#
Debug.Assert(ConfigurationPath.KeyDelimeter == ":");
int indexOf = key.IndexOf(':', prefixLength);
```
That said, all of this is really only reasonable if we're confident we'll never change KeyDelimiter, so that first needs to be confirmed.